### PR TITLE
Update versioning

### PR DESCRIPTION
**Pull request by rezabekf**
*Issue #, if available:*
N/A

*Description of changes:*
This PR improves versioning and release. Pre-commit is updated to the latest versions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,12 +41,13 @@ nag:
...
 # Versioning and releases
+.PHONY: version release
 version: $(VENV_NAME)
 	@$(VENV_NAME)/bin/bumpversion $(part)
 
-release:
+release: # run on main branch only
 	@TAG_VERSION=$(shell bumpversion --dry-run --list .bumpversion.cfg | grep current_version | sed s/'^.*='//); \
-		git push origin "v$${TAG_VERSION}" && git push
+		git tag -a "v$${TAG_VERSION}" -m "" && git push origin "v$${TAG_VERSION}"
```
Comment (ConnorKirk) [inline Makefile:50]: For my own curiousity, why are annotated tags used instead of lightweight tags?

**Review by ConnorKirk — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (rezabekf) [reply Makefile:50]: So this is my personal opinion, and I may be completly wrong here, bot for me:

I use **annotated** tags for releases, just to give it it the little bit more matadata like who/what/why and marked them release ready, if that makes sense.

Comment (rezabekf) [reply Makefile:50]: The **lightweight** tags, I don't use much, sometimes when i want to test something and find it later, I may tag it. So mainly I use lightweight tags for private use.

Comment (ConnorKirk) [reply Makefile:50]: Useful to know. Thanks Franco!

**Pull request merged**